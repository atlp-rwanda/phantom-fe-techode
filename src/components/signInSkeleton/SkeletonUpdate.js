import React from "react";
import SkeletonElement from "./SkeletonElement";

const SkeletonUpdate = () => {
    return ( 
            <div className="lg:flex md:flex h-screen animate-pulse">
                <section className="lg:w-5/12 md:w-7/12 w-full lg:flex md:flex flex lg:flex-col flex-col md:flex-col lg:items-center md:items-center items-center animate-pulse">
                    <SkeletonElement type="title" styles="lg:mt-32 lg:ml-60 md:ml-56 md:mt-24 mt-16"/>
                    <SkeletonElement type="subtitle" styles="lg:mt-2 lg:ml-60 md:ml-56 md:mt-2 mt-2" />
                    <SkeletonElement type="input" styles="lg:mt-2 lg:w-1/2 lg:ml-60 md:ml-56 md:mt-2 mt-2" />
                    <SkeletonElement type="input" styles="lg:mt-2 lg:ml-60 md:ml-52 md:mt-2 mt-2" />
                    <SkeletonElement type="text" styles="lg:mt-2 lg:ml-60 md:ml-56 md:mt-2 mt-2" />
                    <SkeletonElement type="inputbutton" styles="lg:mt-2 lg:ml-60 md:ml-56 md:mt-2 mt-2" />
                    <SkeletonElement type="text" styles="lg:mt-2 lg:ml-60 md:ml-56 md:mt-2 mt-2" />
                </section>
            </div>
     );
}
 
export default SkeletonUpdate;