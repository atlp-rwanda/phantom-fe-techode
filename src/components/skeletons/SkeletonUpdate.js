import React from "react";
import SkeletonElement from "./SkeletonElement";

const SkeletonUpdate = () => {
    return ( 
            <div className="border border-slate-200 rounded-md h-full w-full md:w-2/5 md:mt-10 md:mr-10 animate-pulse">
                <section className="flex items-center justify-center">
                    <SkeletonElement type="avatar" styles="mt-5 w-1/6 h-1/6" />
                </section>
                <section className="flex flex-col items-center mt-5 pl-2 pr-2 md:pl-10 md:pr-10 animate-pulse">
                    <SkeletonElement type="input" styles="mt-1 md:mt-5 md:w-4/6 h-8 md:h-11 md:py-2 md:pl-4 md:pr-3"/>
                    <SkeletonElement type="input" styles="mt-1 md:mt-5 md:w-4/6 h-8 md:h-11 md:py-2 md:pl-4 md:pr-3"/>
                    <SkeletonElement type="input" styles="mt-1 md:mt-5 md:w-4/6 h-8 md:h-11 md:py-2 md:pl-4 md:pr-3"/>
                    <SkeletonElement type="input" styles="mt-1 md:mt-5 md:w-4/6 h-8 md:h-11 md:py-2 md:pl-4 md:pr-3"/>
                    <SkeletonElement type="title" styles="mt-5"/>
                    <SkeletonElement type="input" styles="mt-1 md:mt-5 md:w-4/6 h-8 md:h-11 md:py-2 md:pl-4 md:pr-3"/>
                </section>
            </div>
     );
}
 
export default SkeletonUpdate;