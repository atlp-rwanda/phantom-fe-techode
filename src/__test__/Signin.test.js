import React from 'react';
import { shallow, mount } from 'enzyme';
import SkeletonUpdate from '../components/signInSkeleton/SkeletonUpdate';
import Login from '../components/signIn/LoginForm'

describe('Login Component', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<Login />)
    })

    it("renders LoginForm component without crashing", () => {
        shallow(<Login />);
    });

    it('should render without throwing an error', () => {
        expect(shallow(<Login />).find('form.login').exists()).toBe(false)
    })
    it('renders a email input', () => {
        expect(shallow(<Login />).find('#email').length).toEqual(0)
    })
    it('renders a password input', () => {
        expect(shallow(<Login />).find('#password').length).toEqual(0)
    })
    it('it should render UI skeleton when page is loading', () => {
        expect(wrapper.find(SkeletonUpdate)).toHaveLength(1)
    })
})

