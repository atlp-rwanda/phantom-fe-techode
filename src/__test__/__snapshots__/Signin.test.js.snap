// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> It should match the snapshot 1`] = `
<div
  className="lg:flex md:flex w-screen h-screen overflow-hidden"
  style={
    Object {
      "backgroundImage": "linear-gradient(0deg, rgba(243, 246, 249, 0.1), rgba(243, 246, 249, 0.7)),url(pipes.png)",
      "backgroundRepeat": "repeat",
      "backgroundSize": "100px",
    }
  }
>
  <ToastContainer
    autoClose={5000}
    closeButton={[Function]}
    closeOnClick={true}
    draggable={true}
    draggableDirection="x"
    draggablePercent={80}
    hideProgressBar={true}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    role="alert"
    rtl={false}
    theme="light"
    transition={[Function]}
  />
  <div
    className=" letfSide_bg-color lg:w-5/12 md:w-7/12 flex flex-col p-6 md:p-12 lg:px-24 2xl:p-40 "
  >
    <Link
      className="w-full "
      to="/"
    >
      <span
        className="iconify"
      >
        <img
          alt=""
          src="left-arrow.svg"
        />
      </span>
    </Link>
    <div
      className=" w-full h-screen flex items-center flex-wrap"
    >
      <SkeletonUpdate />
    </div>
  </div>
  <div
    className="bg-no-repeat bg-center bg-cover w-3/5 h-screen relative"
  >
    <figure>
      <img
        alt="Background Image"
        className="lg:block md:block lg:mt-10 lg:h-5/6 md:mt-28 md:h-96 lg:mb-20 mr-auto lg:w-10/12 lg:ml-12 hidden"
        src="Image1.png"
      />
      <figcaption
        className="hidden lg:block md:block lg:absolute md:absolute lg:top-1/2 md:top-64 lg:left-1/2 md:left-1/2 lg:right-0 md:right-0 lg:bottom-0 md:bottom-0"
      >
         
        <h3
          className="text-mainColor font-sans font-bold leading-10 lg:text-4xl md:text-2xl lg:mb-2 md:mb-1 lg:mt-1 md:mt-20 not-italic"
        >
          Fast and easy movement !
        </h3>
        <p
          className="text-secondary-300 font-sans not-italic font-semibold leading-10 md:text-base md:mt-2 lg:mt-4 mt-8"
        >
          Get notified anytime, anywhere bus is.Get notified anytime, anywhere bus is.Get notified anytime, anywhere bus is.Get notified anytime, anywhere bus is.
        </p>
      </figcaption>
    </figure>
  </div>
</div>
`;
